.github/workflows/build-test-deploy.yml

```diff
@@ -1014,63 +1014,83 @@ jobs:
           # nix develop .#sre --show-trace -L --command nixops deploy --dry-run --debug --keep-going --show-trace
-  deploy-picasso-frontend:
-    name: "Firebase Picasso Web App Deploy"
-    needs:
-      - local-integration-tests
-      - unit-tests
-    runs-on: ubuntu-latest
+
+
+  frontend-nix-deploy-firebase:
+    name: "Firebase Frontend App Deploy"
+    needs: common-deps
+    runs-on:
+      - self-hosted
+      - linux
+      - x64
+      - sre
+    concurrency:
+      group: packagecomposablenode-${{ github.ref }}
+      cancel-in-progress: true
+    container:
+      image: niteo/nixpkgs-nixos-22.05:316b762afdb9e142a803f29c49a88b4a47db80ee
     steps:
       - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - run: |
+          echo "experimental-features = nix-command flakes" > /etc/nix/nix.conf
+          echo "sandbox = relaxed" >> /etc/nix/nix.conf
+          echo "narinfo-cache-negative-ttl = 0" >> /etc/nix/nix.conf
+      - uses: cachix/cachix-action@f5f67badd061acb62b5c6e25e763572ca8317004
         with:
-          node-version: 16
-          always-auth: true
-          registry-url: 'https://registry.npmjs.org'
-      - run: yarn install && yarn export --filter=picasso
-        working-directory: frontend
-        env:
-          SUBSTRATE_PROVIDER_URL_KUSAMA_2019: ${{ secrets.SUBSTRATE_PROVIDER_URL_KUSAMA_2019 }}
-          SUBSTRATE_PROVIDER_URL_KUSAMA: ${{ secrets.SUBSTRATE_PROVIDER_URL_KUSAMA }}
-          SUBSTRATE_PROVIDER_URL_KARURA: ${{ secrets.SUBSTRATE_PROVIDER_URL_KARURA }}
-          SUBSQUID_URL: ${{ secrets.SUBSQUID_URL }}
-          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
-      - uses: FirebaseExtended/action-hosting-deploy@0f248036885ae672f78587669fa2923f0ef6cac2
+          skipPush: true
+          installCommand: |
+            nix-channel --add ${{env.NIX_NIXPKGS_CHANNEL }} nixpkgs
+            nix-channel --update
+            nix-env -iA nixpkgs.cachix
+          authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+          name: composable-community
+      - run: |
+          cachix watch-exec --jobs 16 --compression-level 0 composable-community nix -- build .#frontend-static-firebase --no-update-lock-file --show-trace -L
+          mkdir -p firebase/picasso/out/
+          mkdir -p firebase/pablo/out/
+          cp -R result/picasso/* firebase/picasso/out/
+          cp -R result/pablo/* firebase/pablo/out/
+          cp frontend/apps/pablo/.firebaserc firebase/pablo/
+          cp frontend/apps/pablo/firebase.json firebase/pablo/
+          cp frontend/apps/picasso/.firebaserc firebase/picasso/
+          cp frontend/apps/picasso/firebase.json firebase/picasso/
+          nix-env -iA nixpkgs.nodejs
+      - name: Deploy Picasso on preview channel
+        if: steps.branch-name.outputs.current_branch != 'main'
+        uses: FirebaseExtended/action-hosting-deploy@0f248036885ae672f78587669fa2923f0ef6cac2
```

Comment:
Does this always deploy, or only if the frontend hash has changed?

Reply:
Currently seeing firebase deployments happening when I change the README. although not too intrusive, it adds to the noise that each PR has with automated bots.

Reply:
It always deploys. not sure how to check the hash to prevent it from deploying.

Reply:
I think the easiest way is to not implement manual hash comparisons ourselves, but to just leverage github workflow's on: directory system

Reply:
https://stackoverflow.com/questions/63822219/how-to-run-github-actions-workflow-only-if-the-pushed-files-are-in-a-specific-fo?rq=1 use on paths

Reply:
Can we just query nix to see if `nix build fe@main` is different from `nix build fe@pr` and based on that deploy?

Reply:
path dependencies by hand are quite brittle

Reply:
I've updated the PR, please have a look.